src/components/reviews/addReviewForm.jsx

```diff
@@ -6,36 +6,126 @@ class AddReviewForm extends React.Component {
     super(props);
 
     this.state = {
-
+      recommend: null,
+      characteristics: {},
+      summary: "",
+      body: "",
+      photos: [],
+      nickname: "",
+      email: "",
+      valid: true
     };
+    this.onChangeBound = this.onChange.bind(this);
+    this.changeCharacteristicBound = this.changeCharacteristic.bind(this);
+    this.submitBound = this.submit.bind(this);
+  }
 
+  onChange(event) {
+    const name = event.target.name;
+    const value = event.target.value;
+    this.setState({
+      [name]: value
```

Comment:
efficient use of onChange event. Makes it so you can dynamically update which saves code space.